src/main/java/edu/ohio/ais/rundeck/HttpBuilder.java

```diff
@@ -438,9 +439,8 @@ public void setHeaders(String headers, RequestBuilder request){
             map = new HashMap<>();
             Object object = null;
             try {
-                YamlReader reader = new YamlReader(headers);
-                object = reader.read();
-                map = (Map<String,String>) object;
+                Yaml yaml = new Yaml(new Constructor(Map.class));
```

Comment:
I think the recommended use of YAML is using a safe constructor

`
Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()))
`

Reply:
@ltamaster I updated it